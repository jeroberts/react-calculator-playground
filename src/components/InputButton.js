import React from 'react';

const InputButton = (props) => {
    return (
        <button 
            className={getButtonStyle(props.value)} 
            onClick={() => props.onClickHandler(props.value)}>
            {props.value}
        </button>
    );
}

const getButtonStyle = value => {
    if ("=" === value) {
        return "equals-button";
    }
    return "button";
}

export default InputButton;