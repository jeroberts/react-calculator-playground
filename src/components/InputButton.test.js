import React from 'react';
import { shallow } from 'enzyme';
import InputButton from './InputButton';

describe('Testing InputButton component', () => {
   
    const wrapper = shallow(<InputButton />)

    // TODO : Why is snapshot so simple? Look into this...
    test('Test that InputButton component renders', () => {
        expect(wrapper).toMatchSnapshot();
    });
});