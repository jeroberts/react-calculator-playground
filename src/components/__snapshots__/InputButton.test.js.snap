// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing InputButton component Test that InputButton component renders 1`] = `ShallowWrapper {}`;
